test/test_prototype_builtin_datasets.py

```diff
@@ -180,3 +180,27 @@ def test_label_matches_path(self, test_home, dataset_mock, config):
         for sample in dataset:
             label_from_path = int(Path(sample["path"]).parent.name)
             assert sample["label"] == label_from_path
+
+
+@parametrize_dataset_mocks(DATASET_MOCKS["usps"])
+class TestUSPS:
+    def test_label_matches_path(self, test_home, dataset_mock, config):
+        dataset_mock.prepare(test_home, config)
+
+        dataset = datasets.load(dataset_mock.name, **config)
+
+        for sample in dataset:
+            # check if correct keys exist
+            assert "image" in sample
+            assert "label" in sample
+
+            # check if correct instance type
+            assert isinstance(sample["image"], Image)
+            assert isinstance(sample["label"], Label)
+
+            # check is data type is correct
+            assert isinstance(sample["image"].data, torch.FloatTensor)
+            assert isinstance(sample["label"].data, torch.LongTensor)
```

Comment:
`torch.FloatTensor` is only the valid class for [tensors on the CPU](https://pytorch.org/docs/master/tensors.html#data-types). Although this is not the case here, this test would fail if we move the data to GPU. What you actually want to check here is

```suggestion
            # check is data type is correct
            assert sample["image"].dtype is torch.float
            assert sample["label"].dtype is torch.long
```

That being said, I'm not sure why you want to enforce the dtype here? Let's say at some point we decide that all images returned by the datasets should be uint8. This test would fail, but I don't see a reason why the dataset would be invalid with that choice.

Reply:
Noted. I just assumed that the datatype of data would always be fixed. Will remove this test.